eessi/testsuite/tests/apps/MetalWalls.py

```diff
@@ -0,0 +1,125 @@
+
+    valid_systems = ['*']
+    valid_prog_environs = ['default']
+    time_limit = '30m'
+
+    module_name = parameter(find_modules('MetalWalls'))
+    # For now, MetalWalls is being build for CPU targets only
+    # compute_device = parameter([DEVICE_TYPES[CPU], DEVICE_TYPES[GPU]])
+    compute_device = parameter([DEVICE_TYPES[CPU], ])
+
+    @run_after('init')
+    def run_after_init(self):
+        """Hooks to run after the init phase"""
+
+        # Filter on which scales are supported by the partitions defined in the ReFrame configuration
+        hooks.filter_supported_scales(self)
+
+        # Make sure that GPU tests run in partitions that support running on a GPU,
+        # and that CPU-only tests run in partitions that support running CPU-only.
+        # Also support setting valid_systems on the cmd line.
+        hooks.filter_valid_systems_by_device_type(self, required_device_type=self.compute_device)
+
+        # Support selecting modules on the cmd line.
+        hooks.set_modules(self)
+
+        # Support selecting scales on the cmd line via tags.
+        hooks.set_tag_scale(self)
+
+    @run_after('init')
+    def set_tag_ci(self):
+        """Set tag CI on smallest benchmark, so it can be selected on the cmd line via --tag CI"""
+        if self.benchmark_info[0] == 'hackathonGPU/benchmark':
+            self.tags.add(TAGS['CI'])
+            log(f'tags set to {self.tags}')
+
+    @run_after('setup')
+    def run_after_setup(self):
+        """Hooks to run after the setup phase"""
+
+        # Calculate default requested resources based on the scale:
+        # 1 task per CPU for CPU-only tests, 1 task per GPU for GPU tests.
+        # Also support setting the resources on the cmd line.
+        if self.compute_device == DEVICE_TYPES[GPU]:
+            hooks.assign_tasks_per_compute_unit(test=self, compute_unit=COMPUTE_UNIT[GPU])
+        else:
+            hooks.assign_tasks_per_compute_unit(test=self, compute_unit=COMPUTE_UNIT[CPU])
+
+    @run_after('setup')
+    def set_binding(self):
+        """Set binding to compact to improve performance reproducibility."""
+        hooks.set_compact_process_binding(self)
+
+    @run_after('setup')
+    def request_mem(self):
+        """Request memory per node based on the benchmark."""
+        mem_per_task = 0.4
+        if self.benchmark_info[0] == 'hackathonGPU/benchmark5':
+            mem_per_task = 1.2
+        memory_required = self.num_tasks_per_node * mem_per_task + 2
+        hooks.req_memory_per_node(test=self, app_mem_req=memory_required * 1024)
+
+    @run_after('setup')
+    def skip_max_corecnt(self):
+        """Skip tests if number of tasks per node exceeds maximum core count."""
+        max_corecnt = 256
+        bench_name = self.benchmark_info[0]
+        self.skip_if(
+            self.num_tasks > max_corecnt,
+            f'Number of tasks per node {self.num_tasks} exceeds maximum core count {max_corecnt} for {bench_name}'
```

Comment:
Tests are skipped correctly now. I think the message is slightly confusing though. It currently says 'Number of tasks per node', but the relevant thing you limit is actually the _total_ task count (and `self.num_tasks` _is_ the total task count, not the tasks per node).  Also, your calling it 'maximum core count', whereas the thing that gets exceeded is a task count.

I'd phrase it as follows:

```suggestion
            f'Number of tasks {self.num_tasks} exceeds maximum task count {max_task_cnt} for {bench_name}'
```